File: frontend/src/types/link.ts
export type ExternalLinkKind = 'diagnostic' | 'performance' | 'docs' | 'other';
export type ExternalLinkOwnerType = 'simulation' | 'case';

/**
 * Represents a link to an external diagnostic, documentation, or related resource.
 */
export interface ExternalLinkIn {
  kind: ExternalLinkKind;
  url: string;
  label?: string | null;
}

export interface ExternalLinkOut extends ExternalLinkIn {
  id: string; // UUID
  ownerType: ExternalLinkOwnerType;
  createdAt: string; // ISO datetime
  updatedAt: string; // ISO datetime
}

File: backend/app/schemas/link.py
from datetime import datetime
from enum import Enum
from uuid import UUID

from pydantic import Field, HttpUrl

from app.schemas.base import CamelInBaseModel, CamelOutBaseModel


class Kind(str, Enum):
    """Enumeration of possible external link types."""

    DIAGNOSTIC = "diagnostic"
    PERFORMANCE = "performance"
    DOCS = "docs"
    OTHER = "other"


class ExternaLinkCreate(CamelInBaseModel):
    kind: Kind = Field(..., description="The type of the external link.")
    url: HttpUrl = Field(..., description="The URL of the external link.")
    label: str | None = Field(
        None, description="An optional label for the external link."
    )


class ExternalLinkOut(CamelOutBaseModel):
    id: UUID = Field(..., description="The unique identifier of the external link.")

    kind: Kind = Field(..., description="The type of the external link.")
    url: HttpUrl = Field(..., description="The URL of the external link.")
    label: str | None = Field(
        None, description="An optional label for the external link."
    )

    created_at: datetime = Field(
        ..., description="The timestamp when the external link was created."
    )
    updated_at: datetime = Field(
        ..., description="The timestamp when the external link was last updated."
    )
